source/math/FloatMatrix.ooc

```diff
@@ -123,12 +126,13 @@ FloatMatrix : cover {
 	// Tranpose matrix. Creates a new matrix being the transpose of the current matrix.
 	// </summary>
 	// <returns>Return current matrix tranposed.</returns>
-	transpose: func@ -> This {
+	transpose: func -> This {
 		result := This new(this dimensions swap())
 		for (y in 0 .. this height)
 			for (x in 0 .. this width)
 				result elements[y + x * this height] = this elements[x + y * this width]
-		result
+		this free(Owner Callee)
```

Comment:
Actually... it wouldn't have to be square at all. The transposed matrix will need exactly as much memory as the original either way. And, this applies to any of the function in this class (if we can reuse memory, we should). More than half of all `malloc` calls in the project originate from the `FloatMatrix` constructor...